src/sciline/pipeline.py

```diff
@@ -477,10 +488,15 @@ def _get_provider(
                 return provider, bound
             elif len(matches) > 1:
                 raise AmbiguousProvider("Multiple providers found for type", tp)
-        raise UnsatisfiedRequirement("No provider found for type", tp)
+
+        return handler.handle_unsatisfied_requirement(tp), {}
 
     def build(
-        self, tp: Union[Type[T], Item[T]], /, search_param_tables: bool = False
+        self,
+        tp: Union[Type[T], Item[T]],
+        /,
+        handler: Handler,
```

Comment:
Should this have a default value?

Reply:
Not really. Users should be using `get`.

Reply:
Then, should `build` not be public?

Reply:
I think there is some value in having it. And the docstring is explicit about it, and refers to `get`.